addons/mass_mailing/models/mailing.py

```diff
@@ -730,7 +730,8 @@ def _get_default_mailing_domain(self):
 
         if self.mailing_type == 'mail' and 'is_blacklisted' in self.env[self.mailing_model_name]._fields:
             mailing_domain = expression.AND([[('is_blacklisted', '=', False)], mailing_domain])
-        if self.mailing_type == 'mail' and 'opt_out' in self.env[self.mailing_model_name]._fields:
+        if self.mailing_type == 'mail' and self.mailing_model_name != 'mailing.contact' and\
```

Comment:
I think the condition could simply be removed. 
Opt out is a field only on mailing.contact model and cannot be used here. 
It's also a field on mailing.contact.subscription model but it cannot be used either as target model is mailing.list and the opt out is automatically handled during the sending of the mails.
So the condition you wrote will never be reached.